src/helpers.js

```diff
@@ -47,10 +51,34 @@ function typeNameTranslator(lang) {
   })[lang] || function(x) { return x; };
 }
 
-module.exports = function(lang) {
-  return {
-    typeName: typeNameTranslator(lang),
+function paramsForAction(action) {
+    // Figure out how many params will be consumed by the path and put the
+  // first N required params there - the rest go in options.
+  var numPathParams = (action.path.match(/%/g) || []).length;
+  var results = {
+    pathParams: [],
+    explicitNonPathParams: [],
+    optionParams: []
+  };
+  if (action.params) {
+    action.params.forEach(function(param, index) {
+      if (param.required && results.pathParams.length < numPathParams) {
+        results.pathParams.push(param);
+      } else if (param.explicit) {
+        results.explicitNonPathParams.push(param);
+      } else {
+        results.optionParams.push(param);
+      }
+    });
+  }
+  return results;
+}
+
+var langs = {
```

Comment:
Agree, and I actually did that when I was working on the PHP template but forgot to push it. I'll do that right now.